tapir/core/tapir_email_base.py

```diff
@@ -11,43 +11,44 @@
 
 if TYPE_CHECKING:
     # ignore at runtime to avoid circular import
-    from tapir.accounts.models import TapirUser
+    from tapir.accounts.models import TapirUser, OptionalMails
     from tapir.coop.models import ShareOwner
 from tapir.log.models import EmailLogEntry
 
 all_emails: Dict[str, Type[TapirEmailBase]] = {}
 
+MAIL_OPTIONS_ = Literal[True, False, "both"]
 
-def get_all_emails(default: bool | None = True) -> list[Type[TapirEmailBase]]:
-    if default is not None:
-        return [
-            mail
-            for mail in TapirEmailBase.__subclasses__()
-            if mail.enabled_by_default is default
-        ]
-    else:
-        return [mail for mail in TapirEmailBase.__subclasses__()]
 
+def get_mail_types(
+    enabled_by_default: MAIL_OPTIONS_ = True,
+    optional: MAIL_OPTIONS_ = True,
+) -> List[Tuple[str, str]]:
+    """
+    default="both" returns both, default and non-default mails.
+    default=False returns mails not being sent by default
+    """
+
+    def filter_mail(mail):
+        return (
+            enabled_by_default == "both"
+            or mail.enabled_by_default is enabled_by_default
+        ) and (optional == "both" or mail.optional is optional)
 
-def mails_not_mandatory(default: bool | None = True) -> List[Tuple[str, str]]:
     return [
```

Comment:
Is there an advantage to returning the id, name pair here? We could just return the list of classes directly and let the callers of `get_mail_type` then decide what they what to do with it.

Reply:
I thought tuples were necessary for choices, but have to check

Reply:
yes, it was intended for choices but I agree with you to let receivers decide